From: Maximilian Weigand <mweigand@mweigand.net>
Date: Wed, 24 Aug 2022 08:21:02 +0200
Subject: rk356x.dtsi: add entry for the rockchip-rga, disabled.

enable in arch/arm64/boot/dts/rockchip/rk3566-pinenote.dtsi, node rk_rga

todo: the entry does not define resets for the rga as this leads to a
kernel panic. Therefore module unloading does not work at the moment and
the module should be blacklisted to prevent accidental loading.
---
 arch/arm64/boot/dts/rockchip/rk3566-pinenote.dtsi |  4 +++
 arch/arm64/boot/dts/rockchip/rk356x.dtsi          | 30 +++++++++++++++++++++++
 2 files changed, 34 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3566-pinenote.dtsi b/arch/arm64/boot/dts/rockchip/rk3566-pinenote.dtsi
index 8c725d8..3bb38a6 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-pinenote.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3566-pinenote.dtsi
@@ -322,6 +322,10 @@ &gpu {
 	// status = "okay";
 };
 
+&rk_rga{
+       status = "disabled";
+};
+
 &i2c0 {
 	status = "okay";
 
diff --git a/arch/arm64/boot/dts/rockchip/rk356x.dtsi b/arch/arm64/boot/dts/rockchip/rk356x.dtsi
index 85c8b7e..4f28b89 100644
--- a/arch/arm64/boot/dts/rockchip/rk356x.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk356x.dtsi
@@ -643,6 +643,36 @@ vpu: video-codec@fdea0400 {
 		clock-names = "aclk", "hclk";
 		iommus = <&vdpu_mmu>;
 		power-domains = <&power RK3568_PD_VPU>;
+
+	rk_rga: rk_rga@fdeb0000 {
+		compatible = "rockchip,rk356x-rga";
+		reg = <0x0 0xfdeb0000 0x0 0x1000>;
+		interrupts = <GIC_SPI 90 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&cru ACLK_RGA>, <&cru HCLK_RGA>, <&cru CLK_RGA_CORE>;
+		clock-names = "aclk_rga", "hclk_rga", "clk_rga";
+		power-domains = <&power RK3568_PD_RGA>;
+		/* 754 #define SRST_A_RGA      292 */
+		/* 755 #define SRST_H_RGA      293 */
+		/* 756 #define SRST_RGA_CORE       294 */
+		/* resets = <&cru SRST_RGA_CORE>, <&cru SRST_A_RGA>, <&cru SRST_H_RGA>; */
+		/* reset-names = "core", "axi", "ahb"; */
+		// from the documenation:
+		// dma-coherent;
+		status = "disabled";
+	};
+
+	ebc: ebc@fdec0000 {
+		compatible = "rockchip,rk3568-ebc";
+		reg = <0x0 0xfdec0000 0x0 0x5000>;
+		interrupts = <GIC_SPI 17 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&cru HCLK_EBC>, <&cru DCLK_EBC>;
+		clock-names = "hclk", "dclk";
+		pinctrl-0 = <&ebc_pins>;
+		pinctrl-names = "default";
+		power-domains = <&power RK3568_PD_RGA>;
+		resets = <&cru SRST_H_EBC>, <&cru SRST_D_EBC>;
+		reset-names = "hclk", "dclk";
+		status = "disabled";
 	};
 
 	vdpu_mmu: iommu@fdea0800 {
